### PR TITLE
@types/react-native add support for onFocus/onBlur and Platform.isTV

### PR DESCRIPTION
**Pull request by tykus160**
Adding onFocus/onBlur for TouchableWithoutFeedback and Platform.isTV boolean.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://facebook.github.io/react-native/docs/touchablewithoutfeedback#onfocus
https://facebook.github.io/react-native/docs/touchablewithoutfeedback#onblur
https://facebook.github.io/react-native/docs/building-for-apple-tv

### CODE REVIEW

**Review by saranshkataria — APPROVED** [review]

**Review by alloy — COMMENTED** [review]
Why are we using `any` here `NativeSyntheticEvent<any>`, is there nothing more specific?

**Comment by tykus160** [comment]
This event is defined in React Native as follows:
```
type TargetEvent = SyntheticEvent<
  $ReadOnly<{|
    target: number,
  |}>,
>;
```
And to be honest I'm not sure how to properly translate it into typescript.

File: types/react-native/index.d.ts [hunk 1]
```diff
@@ -5169,6 +5169,20 @@ export interface TouchableWithoutFeedbackProps extends TouchableWithoutFeedbackP
      */
     hitSlop?: Insets;
 
+    /**
+     * When `accessible` is true (which is the default) this may be called when
+     * the OS-specific concept of "blur" occurs, meaning the element lost focus.
+     * Some platforms may not have the concept of blur.
+     */
+    onBlur?: (e: NativeSyntheticEvent<any>) => void;
```
Comment (alloy) [inline types/react-native/index.d.ts]:
```suggestion
    onBlur?: (e: NativeSyntheticEvent<{ target: number }>) => void;
```

File: types/react-native/index.d.ts [hunk 2]
```diff
@@ -5169,6 +5169,20 @@ export interface TouchableWithoutFeedbackProps extends TouchableWithoutFeedbackP
...
+
+    /**
+     * When `accessible` is true (which is the default) this may be called when
+     * the OS-specific concept of "focus" occurs. Some platforms may not have
+     * the concept of focus.
+     */
+    onFocus?: (e: NativeSyntheticEvent<any>) => void;
```
Comment (alloy) [inline types/react-native/index.d.ts]:
```suggestion
    onFocus?: (e: NativeSyntheticEvent<{ target: number }>) => void;
```

**Comment by alloy** [comment]
I see, thanks for digging that up! I’ve left some suggestions based on that, which you can commit if you agree.

It seems like we could refactor some of our event data interfaces to all extend a base `TargetedEvent` interface that has just `target: number`, but that doesn’t need to happen now if you don’t feel comfortable enough yet with TS.

**Pull request merged**